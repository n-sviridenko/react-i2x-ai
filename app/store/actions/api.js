import { createAction } from 'redux-actions';

export const REQUEST_FAIL = 'app/api/REQUEST_FAIL';

export const requestFail = createAction(REQUEST_FAIL);
