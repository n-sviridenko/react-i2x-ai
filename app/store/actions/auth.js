import { createAction } from 'redux-actions';
import { Map } from 'immutable';

export const LOGIN_REQUEST = 'app/auth/LOGIN_REQUEST';
export const LOGIN_SUCCESS = 'app/auth/LOGIN_SUCCESS';
export const LOGIN_FAILURE = 'app/auth/LOGIN_FAILURE';

export const loginRequest = createAction(LOGIN_REQUEST, (email, password) => Map({ email, password }));
export const loginSuccess = createAction(LOGIN_SUCCESS);
export const loginFailure = createAction(LOGIN_FAILURE);
