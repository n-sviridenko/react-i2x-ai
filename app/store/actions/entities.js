import { createAction } from 'redux-actions';

export const MERGE = 'app/entities/MERGE';

export const merge = createAction(MERGE);
