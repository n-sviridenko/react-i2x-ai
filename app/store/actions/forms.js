import { createAction } from 'redux-actions';

export const SUBMIT = 'app/forms/SUBMIT';

export const submit = createAction(SUBMIT, () => null, (options) => options);

export function buildSubmit(requestActionCreator, successActionType, failActionType) {
  return (values, dispatch) => new Promise((resolve, reject) => {
    const submitOptions = {
      requestAction: requestActionCreator(values),
      successActionType,
      failActionType,
      resolve,
      reject,
    };

    dispatch(submit(submitOptions));
  });
}
