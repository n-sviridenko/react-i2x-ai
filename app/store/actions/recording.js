import { createAction } from 'redux-actions';

export const LIST_REQUEST = 'app/recording/LIST_REQUEST';
export const LIST_SUCCESS = 'app/recording/LIST_SUCCESS';
export const LIST_FAIL = 'app/recording/LIST_FAIL';

export const listRequest = createAction(LIST_REQUEST);
export const listSuccess = createAction(LIST_SUCCESS);
export const listFail = createAction(LIST_FAIL);
