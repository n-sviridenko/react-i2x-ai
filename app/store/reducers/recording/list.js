import { List } from 'immutable';

import loaderCreator from 'store/reducers/creators/loaderCreator';
import { LIST_REQUEST, LIST_SUCCESS, LIST_FAIL } from 'store/actions/recording';

export default loaderCreator({
  types: [
    LIST_REQUEST,
    LIST_SUCCESS,
    LIST_FAIL,
  ],
  requestPayloadMapper: () => List(),
});

const getRoot = (state) => state.getIn(['recording', 'list']);

export {
  getRoot,
};
