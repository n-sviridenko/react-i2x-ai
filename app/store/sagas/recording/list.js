import { call, put } from 'redux-saga/effects';

import * as actions from 'store/actions/recording';
import sendRequest from 'store/sagas/api/sendRequest';

export default function* list() {
  try {
    const res = yield call(sendRequest, 'get', 'ai/recording/list');

    yield put(actions.listSuccess(res.get('results')));
  } catch (err) {
    yield put(actions.listFail(err));
  }
}
