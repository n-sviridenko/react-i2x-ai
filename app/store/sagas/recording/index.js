import { takeLatest } from 'redux-saga/effects';

import * as actions from 'store/actions/recording';
import list from './list';

export function* listWatcher() {
  yield takeLatest(actions.LIST_REQUEST, list);
}

export default [
  listWatcher,
];
