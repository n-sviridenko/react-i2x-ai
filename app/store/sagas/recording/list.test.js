/* eslint-disable redux-saga/yield-effects */
import { put } from 'redux-saga/effects';
import { Map, List } from 'immutable';

import * as actions from 'store/actions/recording';
import list from './list';

describe('list', () => {
  let listTask;

  beforeEach(() => {
    listTask = list();

    const callDescriptor = listTask.next().value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it('should dispatch the listSuccess if request was succeeded', () => {
    const results = List();
    const res = Map({ count: 0, next: null, previous: null, results });
    const putDescriptor = listTask.next(res).value;
    expect(putDescriptor).toEqual(put(actions.listSuccess(results)));
  });

  it('should dispatch the listFail if request was failed', () => {
    const error = new Error();
    const putDescriptor = listTask.throw(error).value;
    expect(putDescriptor).toEqual(put(actions.listFail(error)));
  });
});
