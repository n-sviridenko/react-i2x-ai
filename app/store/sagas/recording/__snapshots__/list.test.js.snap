// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list should dispatch the listFail if request was failed 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "get",
      "ai/recording/list",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`list should dispatch the listSuccess if request was succeeded 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "get",
      "ai/recording/list",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
